### PR TITLE
Added USB interface association descriptor for GD32VF103.

### PR DESCRIPTION
**Pull request by michahoiting**
I am working on a private project using the GD32VF103 and implementing an ACM-CDC USB device. This profile needs the *USB interface association descriptor* to work properly. In this pull request, I have added this descriptor for the GD32VF103 in the folder `nuclei-sdk/SoC/gd32vf103/Common/Include/Usb/`. See also my use case: https://github.com/michahoiting/rv-link.

Kind regards,
Micha

### CODE REVIEW

File: SoC/gd32vf103/Common/Include/Usb/usb_ch9_std.h [hunk 1]
```diff
@@ -98,6 +99,7 @@ enum _usb_desctype {
     USB_DESCTYPE_DEV_QUALIFIER    = 0x6U,        /*!< USB device qualtfier descriptor type */
     USB_DESCTYPE_OTHER_SPD_CONFIG = 0x7U,        /*!< USB other speed configuration descriptor type */
     USB_DESCTYPE_ITF_POWER        = 0x8U,        /*!< USB interface power descriptor type */
+    USB_DESCTYPE_ITF_ASSOCIATION  = 0xbU,        /*!< USB interface association descriptor type */
```
Comment (michahoiting) [inline SoC/gd32vf103/Common/Include/Usb/usb_ch9_std.h:102]: Added according to the document [USB Interface Association Descriptor Device Class Code and Use Model](https://usb.org/sites/default/files/iadclasscode_r10.pdf) (Revision 1.0).

File: SoC/gd32vf103/Common/Include/Usb/usb_ch9_std.h [hunk 2]
```diff
@@ -213,6 +215,17 @@ typedef struct _usb_desc_itf {
     uint8_t iInterface;                   /*!< index of the string descriptor describing the interface */
 } usb_desc_itf;
 
+typedef struct _usb_desc_itf_association {
```
Comment (michahoiting) [inline SoC/gd32vf103/Common/Include/Usb/usb_ch9_std.h:218]: Added according to the document [USB ENGINEERING CHANGE NOTICE](http://www.skembedded.com/usb_spec/usb_20/InterfaceAssociationDescriptor_ecn.pdf).

File: SoC/gd32vf103/Common/Include/Usb/usb_ch9_std.h (hunk 1 above)
Comment (michahoiting) [reply SoC/gd32vf103/Common/Include/Usb/usb_ch9_std.h]: Is corrected to upper-case 0xBU.

**Comment by fanghuaqi** [comment]
Hi @michahoiting , did you tested your branch with the changes? Will it affect other target boards?

**Comment by michahoiting** [comment]
Hi @fanghuaqi, for my project using this branch, I successfully build my application for the GD32VFC-START and Sipeed Longan Nano boards, see also: https://travis-ci.com/github/michahoiting/rv-link. I only have Longan Nano boards, and on that board it runs OK.
Also, this PR only brings one new a define and struct typdef, so this is very unlikely to break builds.

**Comment by fanghuaqi** [comment]
Ok, I will just squash it into 1 commit and merged it

**Comment by michahoiting** [comment]
I just did a build in the SDK according to the README.md.
- make PROGRAM=application/baremetal/helloworld all -> OK
- make PROGRAM=application/baremetal/helloworld SOC=gd32vf103 all -> OK

**Pull request merged**